internal/k8s/appprotectdos/app_protect_dos_resources.go

```diff
@@ -0,0 +1,151 @@
+package appprotectdos
+
+import (
+	"fmt"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var appProtectDosPolicyRequiredFields = [][]string{
+	{"spec", "mitigation_mode"},
+	{"spec", "automation_tools_detection"},
+	{"spec", "tls_fingerprint"},
+	{"spec", "signatures"},
+	{"spec", "bad_actors"},
+}
+
+var appProtectDosLogConfRequiredFields = [][]string{
+	{"spec", "content"},
+	{"spec", "filter"},
+}
+
+func validateRequiredFields(policy *unstructured.Unstructured, fieldsList [][]string) error {
```

Comment:
in this file we have multiple functions similar to the functions in app_protect_configuration.go
- validateRequiredFields
- ValidateAppProtectDosLogDestination (similar to ValidateAppProtectLogDestination)
- GetNsName 
- ParseResourceReferenceAnnotation

I suggest adding another package - `appprotect_common` and move common functions there and reference them from the `approtect` and `approtectdos` packages.

Later, after this PR, we can further refactor the package structure - for example, have `appprotect`, `approtect/waf`, `approtect/dos`, `appprotect/common` packages. However, we don't need to do it now, so that we don't increase the amount of changes in this PR.

Reply:
Add package appprotect_common